src/main/java/com/openlattice/edm/PostgresEdmManager.java

```diff
@@ -173,20 +119,29 @@ public void createEntitySet( EntitySet entitySet, Collection<PropertyType> prope
         }
     }
 
-    public Iterable<PropertyUsageSummary> getPropertyUsageSummary( String propertyTableName ) {
-        String getPropertyTypeSummary =
-                String.format(
-                        "SELECT %1$s , %6$s AS %2$s , %3$s, COUNT(*) FROM %4$s LEFT JOIN %5$s ON %3$s = %5$s.id AND %4$s.version > 0 GROUP BY ( %1$s , %2$s, %3$s )",
-                        ENTITY_TYPE_ID_FIELD,
-                        ENTITY_SET_NAME_FIELD,
-                        ENTITY_SET_ID_FIELD,
-                        propertyTableName,
-                        ENTITY_SETS,
-                        NAME_FIELD );
+    public Iterable<PropertyUsageSummary> getPropertyUsageSummary( UUID propertyTypeId ) {
+        final var wrappedEntitySetsTableName = "wrapped_entity_sets";
+        final var getPropertyTypeSummary =
+                String.format( "WITH %1$s AS (SELECT %2$s, %3$s AS %4$s, %5$s FROM %6$s) " +
+                                "SELECT %2$s, %4$s, %7$s, COUNT(*) FROM %8$s LEFT JOIN %1$s ON %7$s = %1$s.id " +
+                                "WHERE %9$s > 0 AND %10$s = ? GROUP BY ( %2$s , %4$s, %7$s )",
```

Comment:
this hurts my brain just a little bit

Reply:
Yeah, it was really fun to check the order :D